x/concentrated-liquidity/position.go

```diff
@@ -621,17 +610,54 @@ func (k Keeper) RemovePositionIdToLock(ctx sdk.Context, positionId, underlyingLo
 	store.Delete(lockIdKey)
 }
 
-// positionHasUnderlyingLockInState checks if a given positionId has a corresponding lock in state.
-func (k Keeper) positionHasUnderlyingLockInState(ctx sdk.Context, positionId uint64) (bool, error) {
+// PositionHasActiveUnderlyingLock is a non mutative method that checks if a given positionId has a corresponding lock in state.
+// If it has a lock in state, checks if that lock is still active.
+// If lock is still active, returns true.
+// If lock is no longer active, returns false.
+func (k Keeper) PositionHasActiveUnderlyingLock(ctx sdk.Context, positionId uint64) (hasActiveUnderlyingLock bool, lockId uint64, err error) {
```

Comment:
nit: I think non-mutative version doesn't have to be exported

Reply:
I think you meant the opposite, the mutative version does not need to be exported. Done here https://github.com/osmosis-labs/osmosis/pull/5042/commits/e3212b2d10f08587f1c9ea958875b641c4363427